### DISCUSSION TITLE
A workaround for the removed ForAllOtherMembers(opt => opt.Ignore())

### DISCUSSION THREAD

**Discussion by wertzui** [LuckyPennySoftware/AutoMapper#3911] [Show and tell]
As already described in https://github.com/AutoMapper/AutoMapper/discussions/3833 ForAllOtherMembers was removed in the recent version of AutoMapper.

I too was relying on it to map only specific properties of a source into an existing destination object.

@jakubfijalkowski made a great example in that discussion:
```
public class MyDto
    {
        public Guid Id { get; set; }

        public string A { get; set; }

        public string B { get; set; }
    }

    public class MyA
    {
        public Guid Id { get; set; }

        public string A { get; set; }
    }

    public class MyB
    {
        public Guid Id { get; set; }

        public string B { get; set; }
    }

CreateMap<MyA, MyDto>();
CreateMap<MyB, MyDto>()
    .ForMember(dst => dst.B, opt => opt.MapFrom(src => src.B))
    .ForAllOtherMembers(opt => opt.Ignore()); // No longer possible
```

So instead of listing every wanted member with `ForMember` and ignoring the rest with `ForAllOtherMembers`, it is possible to achieve a similar outcome using this code:
```
CreateMap<MyB, MyDto>()
	.ConvertUsing((src, dst) => 
	{
		dst.B = src.B;  // or dst.B = ctx.Mapper.Map<string>(src.B); if you need other configured mappings there.
		return dst;
	});
```

It is not much harder to write and you can keep your existing code that looks like this
```
var dto = mapper.Map<MyDto>(a);
mapper.Map(b, dto);
```

I know that is does not work together with EntityFramework, or anything else that relies on analyzing the expressions.
However for simple scenarios this might save some people rewriting their business code.

**Comment by thesushil** [comment]
@wertzui Can you please explain why your solution doesn't work with EntityFramework? Would it work in ef core?

**Reply by wertzui** [reply]
I't won't work with either, because it is calling `ConvertUsing` which does not get translated into SQL, because it is no `Expression`, but a `Func`.

**Reply by lbargaoanu** [reply]
@wertzui A cleaner way to write this mapping is with `IncludeMembers`.

**Comment by thesushil** [comment]
In my case it threw null reference while assigning the first member.
I had to create the destination object like below:
```
CreateMap<MyB, MyDto>()
	.ConvertUsing((src, dst) => 
	{
               dst = dst ?? new MyDto();
		dst.B = src.B;  // or dst.B = ctx.Mapper.Map<string>(src.B); if you need other configured mappings there.
		return dst;
	});
```
